db/sql/0_msar.sql

```diff
@@ -1528,3 +1588,100 @@ BEGIN
   RETURN created_table_id;
 END;
 $$ LANGUAGE plpgsql;
+
+
+----------------------------------------------------------------------------------------------------
+----------------------------------------------------------------------------------------------------
+-- MATHESAR LINK FUNCTIONS
+--
+-- Add a link to the table.
+----------------------------------------------------------------------------------------------------
+----------------------------------------------------------------------------------------------------
+
+-- Create a Many-to-One or a One-to-One link -------------------------------------------------------
+
+
+CREATE OR REPLACE FUNCTION
+msar.create_many_to_one_link(
+  from_rel_id oid,
+  to_rel_id oid,
+  col_name text,
+  unique_link boolean DEFAULT false
+) RETURNS smallint AS $$/* 
+Create a many-to-one or a one-to-one link between tables, returning the attnum of the newly created
+column.
+
+Args:
+  from_rel_id: The OID of the referent table.
+  to_rel_id: The OID of the referrer table.
+  col_name: Name of the new column to be created in the referrer table, unqoted. 
+  unique_link: Whether to make the link one-to-one instead of many-to-one.
+*/
+DECLARE
+  pk_col_id smallint;
+  pk_col_type text;
+  col_defs jsonb;
+  added_col_id smallint[];
+  uq_con_def jsonb;
+  fk_con_def jsonb;
+BEGIN
+  pk_col_id := msar.get_pk_column(from_rel_id);
+  pk_col_type := msar.get_column_type(from_rel_id, pk_col_id);
+  col_defs := format(
+    '[{
+        "name": "%s",
+        "type": {"name": "%s"}
```

Comment:
I am a bit uncomfortable with using double quotes here as it _will_ fail if the column names have a double quote in them but this is the best way I found to get it working. The problem was that even after using `quote_ident` if the column name didn't have any spaces they didn't get quoted and `quote_literal` quotes the names with single quotes which is a problem when casting to jsonb, same with `quote_ident` if the col name isn't quoted we can't cast the object to jsonb.